test/cpp/api/modules.cpp

```diff
@@ -2079,3 +2079,11 @@ TEST_F(ModulesTest, PrettyPrintThreshold) {
       ThresholdOptions(42.42, 24.24).inplace(true))),
     "torch::nn::Threshold(threshold=42.42, value=24.24, inplace=true)");
 }
+
+TEST_F(ModulesTest, PrettyPrintAlphaDropout) {
+  ASSERT_EQ(c10::str(AlphaDropout(0.2)),
+    "torch::nn::AlphaDropout(rate=0.2)");
+  ASSERT_EQ(c10::str(AlphaDropout(
+      DropoutOptions(0.2))),
+    "torch::nn::AlphaDropout(rate=0.2)");
+}
```

Comment:
It would be awesome to add a module test similar to
https://github.com/pytorch/pytorch/blob/e31adeb4f38b41de5c0f840b6cdf05593711d4da/test/cpp/api/modules.cpp#L897-L911

Reply:
I have added a test similar to the one for `Dropout`. From what I read in [this](https://arxiv.org/pdf/1706.02515.pdf) paper, the sum of the output should lie between 40 and 130

>New Dropout Technique. Standard dropout randomly sets an activation x to zero with probability 1 − q for 0 < q 6 1 .  In order to preserve the mean, the activations are scaled by 1 /q during training.  If x has mean E(x) = μ and variance Var(x) = ν , and the dropout variable d follows a binomial distribution B(1 ,q ) , then the mean E(1 /qdx ) = μ is kept.  Dropout fits well to rectified linear units, since zero is in the low variance region and corresponds to the default value. For scaled exponential linear units, the default and low variance value is lim x →−∞ selu(x)  = − λα = α ′ . Therefore, we propose “alpha dropout”, that randomly sets inputs to α ′ .  The new mean and new variance is E(xd + α ′(1 − d)) = qμ + (1 − q)α ′ , and Var(xd + α ′(1 − d)) = q((1 − q)(α ′ − μ) 2 + ν) .  We aim at keeping mean and variance to their original values after “alpha dropout”, in order to ensure the self-normalizing property even for “alpha dropout”. The affine transformation a (xd + α ′ (1 − d)) + b allows to determine parameters a and b such that mean and variance are kept to their values: E( a ( xd + α ′ (1 − d )) + b ) = μ and Var( a ( xd + α ′ (1 − d )) + b ) = ν . In contrast to dropout, a and b will depend on μ and ν , however our SNNs converge to activations with zero mean and unit variance.  With μ = 0 and ν = 1 , we obtain a = (q + α′2q(1 − q )) − 1 / 2 and b = − ( q + α ′ 2 q (1 − q ) ) − 1 / 2 ((1 − q ) α ′ ) . The parameters a and b only depend on the dropout rate 1 − q and the most negative activation α

Reply:
Thanks so much for adding the test! :D